modules/site-health/admin-site-health.php

```diff
@@ -0,0 +1,272 @@
+	 */
+	public function info_languages( $debug_info ) {
+		$languages = PLL()->model->get_languages_list();
+		$fields = array();
+		foreach ( $languages as $language ) {
+			foreach ( $language as $key => $value ) {
+				if ( empty( $value ) ) {
+					$value = '0';
+				}
+
+				if ( in_array(
+					$key,
+					$this->exclude_lang_key()
+				)
+				) {
+					continue;
+				}
+
+				$fields[ $key ]['label'] = $key;
+				$fields[ $key ]['value'] = $value;
+
+				if ( 'term_group' === $key ) {
+					$fields[ $key ]['label'] = _x( 'order', 'Order of the language in the language switcher', 'polylang' );
+				}
+			}
+			$flag = $this->get_flag( $language );
+
+			$lang_name = sanitize_title( $language->name );
+			$debug_info[ $lang_name ] = array(
+				// translators: placeholder is the language name
+				'label'  => sprintf( __( 'Language: %s', 'polylang' ), $language->name ),
+				// translators: placeholder is the flag image
+				'description' => sprintf( __( 'Flag used in the language switcher: %s', 'polylang' ), $flag ),
+				'fields' => $fields,
+			);
+		}
+		return $debug_info;
+	}
+
+	/**
+	 * Return the flag used in the language switcher
+	 *
+	 * @param object $language language object
+	 * @return mixed
+	 * @since   2.8
+	 */
+	public function get_flag( $language ) {
+		if ( empty( $language->flag ) ) {
+			return $flag = '<span class="no-flag">' . __( 'Undefined', 'polylang' ) . '</span>';
+		}
+		if ( ! empty( $language->custom_flag_url ) ) {
+			return $flag = '<img src="' . $language->custom_flag_url . '" height="11">';
+		} else {
+			return $flag = $language->flag;
+		}
+	}
+
+	/**
+	 * Add a Site Health test on Home Page translation
+	 *
+	 * @param array $tests array with tests declaration data
+	 * @return array
+	 * @since   2.8
+	 */
+	public function is_homepage( $tests ) {
+		// add test only if static page on front page.
+		if ( '0' !== get_option( 'page_on_front' ) ) {
+			$tests['direct']['pll_homepage'] = array(
+				'label' => __( 'Home page translated', 'polylang' ),
+				'test'  => array( $this, 'homepage_test' ),
+			);
+		}
+		return $tests;
+	}
+
+	/**
+	 * Test if the home page is translated or not.
+	 *
+	 * @return array $result array with test results
+	 * @since 2.8
+	 */
+	public function homepage_test() {
+		$result = array(
+			'label'       => __( 'All languages have a translated home page', 'polylang' ),
+			'status'      => 'good',
+			'badge'       => array(
+				'label' => __( 'Polylang', 'polylang' ),
+			),
+			'description' => sprintf(
+				'<p>%s</p>',
+				__( 'A website can\'t be displayed without homepage.', 'polylang' )
+			),
+			'actions'     => '',
+			'test'        => 'pll_homepage',
+		);
+		$untranslated = array();
+		foreach ( PLL()->model->get_languages_list() as $language ) {
+			if ( ! $language->page_on_front ) {
+				$untranslated[] = sprintf(
+					'<a href="%s">%s</a>',
+					esc_url( PLL()->links->get_new_post_translation_link( get_option( 'page_on_front' ), $language ) ),
```

Comment:
You can have the links as a dependency of this class in the same way than you did for the `PLL_Model`.